website/project/model.py

```diff
@@ -75,16 +74,9 @@ def has_anonymous_link(node, auth):
     :param str link: any view-only link in the current url
     :return bool anonymous: Whether the node is anonymous to the user or not
     """
-    view_only_link = auth.private_key or request.args.get('view_only', '').strip('/')
-    if not view_only_link:
-        return False
-    if node.is_public:
-        return False
-    return any(
-        link.anonymous
-        for link in node.private_links_active
-        if link.key == view_only_link
-    )
```

Comment:
This looks like a nice refactoring. I'm working it through to see if the logic is indeed the same. This is where a more complete test suite for `has_anonymous_link` would be helpful (cf. https://github.com/CenterForOpenScience/osf.io/pull/4509/files#r48618100).